### ISSUE TITLE
RFE: Add DNS functionality

### ISSUE THREAD

**Issue by nodomain** [mailcow/mailcow-dockerized#438]
I stumbled over your project by accident and I am currently evaluating to switch from Mailinabox. I am impressed by the functionality and featureset and the use of docker. 
But in my opinion there is a feature missing: Mailinabox provides a DNS functionality where you can be your own primary DNS. I like that feature quite much and would like to discuss the addition of such a feature to Mailcow as well. 

What are your views?

**Comment by mkuron** [comment]
I think an integrated primary DNS server is typically not needed as almost all domain providers allow you to create the necessary records (A, AAAA, CNAME, SRV, TLSA) on their nameservers.

Furthermore, using DNSSEC signing with your own nameserver is difficult as many domain providers do not let you create the DS records needed to publish your public key as a glue record in the parent zone.

**Comment by bronson** [comment]
Interesting, I'm also a mail-in-a-box user.  Its hands-off DNS is one of my favorite features.  I don't miss hand-copying records into grotty web interfaces at all.

All the registrars I've used (6?) have all allowed me to set the DNSSEC glue record.  But I'm sure there are regional differences here.  Maybe I've just been lucky.

So, while your decision makes perfect sense, I look forward to the day that DNSSEC is popular enough to justify automating mailcow's DNS.  :)

**Comment by pixelyo** [comment]
yep, I wish to have this feature as well, I am feeling very uncomfortable managing my own DNS settings, I am currently also using Mail-in-a-Box and this is holding me off switching to mailcow :/

**Comment by labor4** [comment]
Me wishes this too,
I find it very convenient not having to compare records in two ways but just AXFR it to the slave DNS.
Also, the creation process is much faster.
And I'm impressed too! :-)
Coming from Virtualmin, which is a bit overkill for just mail.

**Comment by strarsis** [comment]
Also this can be used for [rotating the DKIM keys](https://support.duocircle.com/support/solutions/articles/5000717184-rotating-your-dkim-key).

It would find this feature really cool!

**Comment by Adorfer** [comment]
I stand by wish to have mailcow integrating a full blown DNS (which can perhaps zonetransfer to services like HurricaneElectric... he.net), since most DNS requirements for the domains i operate are just for mailcow. 
All those text fields etc.... the 2-3 lines for the webserver on the same domain: is just a sidenote....

**Comment by strarsis** [comment]
@Adorfer: Good point, even if the registrar is not flexible enough, 
there are enough free DNS servers that can be used.

**Comment by lenucksi** [comment]
I'd appreciate automated DNS handling too. Some DNS providers offer an API for fully automated DNS updates (Web, not zone transfer) - maybe adding a plugin mechanism for people to provide implementations for their providers' API could be something that might be an approach to this topic.
What do you think?

**Comment by 8ear** [comment]
Perhaps instead of adding DNS functionality in Mailcow check out adguard for this. I think adguard has already an API available.

**Comment by markovendelin** [comment]
Would be great to have that possibility in Mailcow as well. Mailinabox does make it well and simplifies rolling it out greatly. Automated key rotation is also helpful. 

Even if such DNS integration is not possible or has low priority, maybe it is possible / easy to make generator for DNS zonefile files with all recommended settings filled in with the comments? Again, Mailinabox does provide zonefile for copy-and-paste from their UI.
